src/memex/storage.py

```diff
@@ -120,15 +120,17 @@ def create_annotation(request, data):
     annotation = models.Annotation(**data)
     annotation.created = created
     annotation.updated = updated
-    request.db.add(annotation)
 
-    models.update_document_metadata(
+    document = models.update_document_metadata(
         request.db,
         annotation.target_uri,
         document_meta_dicts,
         document_uri_dicts,
         created=created,
         updated=updated)
+    annotation.document_id = document.id
+
+    request.db.add(annotation)
```

Comment:
The next PR will require the `document_id` to be set. And the call to `update_document_metadata` invokes at least one flush which then creates the annotation in the database with `document_id=NULL`.